typescript/scripting-api/index.d.ts

```diff
@@ -6,11 +6,30 @@ type DeepPartial<T> = T extends object ? {
 declare namespace WoT {
 
     /**
-     * Starts the discovery process that will provide ConsumedThing 
-     * 
+     * Starts the discovery process that will provide {@link ThingDescription}
+     * objects for Thing Descriptions that match an optional {@link filter}
+     * argument of type {@link ThingFilter}.
+     *
+     * @param filter represents the constraints for discovering Things as key-value pairs
+     */
+    export function discover(filter?: ThingFilter): ThingDiscoveryProcess;
```

Comment:
I think it needs to be the returning a Promise 

```suggestion
    export function discover(filter?: ThingFilter): Promise<ThingDiscoveryProcess>;
```